src/DynamoCoreWpf/Views/Search/LibraryView.xaml.cs

```diff
@@ -180,19 +184,48 @@ private void ExpandCategory(TreeViewItem sender, NodeCategoryViewModel selectedC
             foreach (var categoryToBeCollapsed in allExpandedCategories)
             {
                 // If class button was clicked.
-                if (categoryToBeExpanded == null)
+                if (selectedClass.SubCategories.Count == 0)
```

Comment:
Sorry, I think I had mistaken it for something else. I have created a new task for you to work on:

[MAGN-6755](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-6755) Simplify the way class button on library is determined